modules/appeals_api/spec/models/supplemental_claim_spec.rb

```diff
@@ -49,163 +50,248 @@
 
-    describe '#city' do
-      subject { supplemental_claim_v2.veteran.city }
+      sc = AppealsApi::SupplementalClaim.new(auth_headers: default_auth_headers, form_data: form_data)
 
-      it { expect(subject).to eq('New York') }
+      expect(sc.stamp_text).to eq 'AAAAAAAAAAbbbbbbbbbbCCCCCCCCCCdd... - 6789'
     end
+  end
 
-    describe '#state_code' do
-      subject { supplemental_claim_v2.veteran.state_code }
+  describe '#update_status!' do
+    let(:supplemental_claim) { create(:supplemental_claim) }
 
-      it { expect(subject).to eq('NY') }
+    it 'error status' do
+      supplemental_claim.update_status!(status: 'error', code: 'code', detail: 'detail')
+
+      expect(supplemental_claim.status).to eq('error')
+      expect(supplemental_claim.code).to eq('code')
+      expect(supplemental_claim.detail).to eq('detail')
     end
 
-    describe '#country_code' do
-      subject { supplemental_claim_v2.veteran.country_code }
+    it 'other valid status' do
+      supplemental_claim.update_status!(status: 'success')
 
-      it { expect(subject).to eq('US') }
+      expect(supplemental_claim.status).to eq('success')
     end
 
-    describe '#zip_code_5' do
-      subject { supplemental_claim_v2.veteran.zip_code_5 }
+    it 'invalid status' do
+      expect do
+        sc_with_nvc.update_status!(status: 'invalid_status')
+      end.to raise_error(ActiveRecord::RecordInvalid,
+                         'Validation failed: Status is not included in the list')
+    end
 
-      it { expect(subject).to eq('30012') }
+    it 'emits events with expected values' do
+      Timecop.freeze(Time.zone.now) do
+        sc_with_nvc.update_status!(status: 'submitted')
+
+        expect(AppealsApi::EventsWorker.jobs.size).to eq(2)
+
+        status_event = AppealsApi::EventsWorker.jobs.first
+        expect(status_event['args']).to eq([
+                                             'sc_status_updated',
+                                             {
+                                               'from' => 'pending',
+                                               'to' => 'submitted',
+                                               'status_update_time' => Time.zone.now.iso8601,
+                                               'statusable_id' => sc_with_nvc.id
+                                             }
+                                           ])
+
+        email_event = AppealsApi::EventsWorker.jobs.last
+        expect(email_event['args']).to eq([
+                                            'sc_received',
+                                            {
+                                              'email_identifier' => {
+                                                'id_type' => 'email',
+                                                'id_value' => 'joe@email.com'
+                                              },
+                                              'first_name' => 'Jäñe',
```

Comment:
appears to be correct: when nvc present, the email event takes in the veteran's first name, but the claimant's email (and first name as well).